FlowCrypt/Controllers/SideMenu/Main/SideMenuNavigationController.swift

```diff
@@ -96,6 +103,167 @@ final class SideMenuNavigationController: ENSideMenuNavigationController {
     }
 }
 
+// MARK: Refresh keys from EKM
+extension SideMenuNavigationController {
+    private func refreshKeysFromEKMIfNeeded(context: AppContextWithUser) {
+        Task {
+            do {
```

Comment:
I'm unsure if this is the best place to put the logic - maybe @sosnovsky will have some thoughts

Reply:
Yeah, I thought about this too and couldn't find better place

Reply:
At a minimum, it could be a separate file? `SideMenuNavigationController+RefreshAccountKeys.swift` ?

Reply:
Sure. Let's see if @sosnovsky has better suggestion

Reply:
For me it'll be better to have separate `EKMService` which will handle all refresh logic, so view controller will just call refresh method like `ekmService.refreshKeysIfNeeded(in: self)`

Also splitting big `refreshKeysFromEKMIfNeeded` method to smaller ones will make it more readable, like this:
```swift
let configuration = try await context.clientConfigurationService.configuration

guard configuration.checkUsesEKM() == .usesEKM else { return }

let emailKeyManagerApi = EmailKeyManagerApi(clientConfiguration: configuration)
let idToken = try await IdTokenUtils.getIdToken(userEmail: context.user.email)
let keyDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
let localKeys = try context.encryptedStorage.getKeypairs(by: context.user.email)

let passPhrase = try fetchCurrentPassphrase()
let keysToUpdate = findExpiredKeys(from: keyDetails)

try await update(keys: keysToUpdate)
showToast("refresh_key_success".localized)
```

Reply:
Nice suggestion

Reply:
will follow your way